src/player.js

```diff
@@ -1288,6 +1299,30 @@ export default class Player extends FakeEventTarget {
     return false;
   }
 
+  /**
+   * update the ABR restriction config
+   * @function _applyABRRestriction
+   * @returns {void}
+   * @param {Object} config - new config which configure for checking if it relevant config has changed
+   * @private
+   */
+  _applyABRRestriction(config: Object): void {
+    if (Utils.Object.hasPropertyPath(config, 'abr.restrictions') && this._engine) {
+      const {restrictions} = this._config.abr;
+      const videoTracks = this._allTracks.filter(track => track instanceof VideoTrack);
+      const newVideoTracks = filterTracksByRestriction(videoTracks, restrictions);
+      if (newVideoTracks.length) {
+        const currentVideoTracks = this._availableTracks.filter(track => track instanceof VideoTrack);
+        this._engine.applyABRRestriction(restrictions);
```

Comment:
this line should be in the if below no?
also `this._engine` could be null this moment

Reply:
engine checked previously